Src/AutoFixture.xUnit.net.UnitTest/FrozenAttributeTest.cs

```diff
@@ -82,18 +69,129 @@ public void GetCustomizationWithSpecificRegisteredTypeReturnsCorrectResult()
+            var seedRequestMatcher = matcher.Specifications.OfType<SeedRequestSpecification>().SingleOrDefault();
+            Assert.NotNull(seedRequestMatcher);
+            Assert.Equal(parameter.ParameterType, seedRequestMatcher.TargetType);
+            // Teardown
+        }
+
+        [Fact]
+        public void GetCustomizationWithMatchingByDirectBaseTypeShouldMatchByBaseType()
+        {
+            // Fixture setup
+            var sut = new FrozenAttribute(Matching.DirectBaseType);
+            // Exercise system
+            var customization = sut.GetCustomization(AParameter<object>());
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<DirectBaseTypeSpecification>());
+            // Teardown
+        }
+
+        [Fact]
+        public void GetCustomizationWithMatchingByImplementedInterfacesShouldMatchByImplementedInterfaces()
+        {
+            // Fixture setup
+            var sut = new FrozenAttribute(Matching.ImplementedInterfaces);
+            // Exercise system
+            var customization = sut.GetCustomization(AParameter<object>());
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<ImplementedInterfaceSpecification>());
+            // Teardown
+        }
+
+        [Fact]
+        public void GetCustomizationWithMatchingByParameterNameShouldMatchByParameter()
+        {
+            // Fixture setup
+            var sut = new FrozenAttribute(Matching.ParameterName, "parameter");
+            // Exercise system
+            var customization = sut.GetCustomization(AParameter<object>());
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<ParameterSpecification>());
+            // Teardown
+        }
+
+        [Fact]
+        public void GetCustomizationWithMatchingByPropertyNameShouldMatchByProperty()
+        {
+            // Fixture setup
+            var sut = new FrozenAttribute(Matching.PropertyName, "Property");
+            // Exercise system
+            var customization = sut.GetCustomization(AParameter<object>());
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<PropertySpecification>());

```

Comment:
The `TargetType` and `TargetName` properties in `PropertySpecification` were deprecated in 088862d so we only look at the type.